README.md
# KeyCode Constants

[![npm version](https://img.shields.io/npm/v/keycode-js.svg?style=flat-square)](https://www.npmjs.com/package/keycode-js)
[![deno](https://img.shields.io/badge/deno-v3.1.0-green?style=flat-square)](https://deno.land/x/keycode/README.md#usagewithdeno)
[![Build Status](https://img.shields.io/travis/kabirbaidhya/keycode-js.svg?style=flat-square)](https://travis-ci.org/kabirbaidhya/keycode-js)
[![Language grade: JavaScript](https://img.shields.io/lgtm/grade/javascript/g/kabirbaidhya/keycode-js.svg?style=flat-square&logo=lgtm&logoWidth=18)](https://lgtm.com/projects/g/kabirbaidhya/keycode-js/context:javascript)
[![npm downloads per week](https://img.shields.io/npm/dw/keycode-js.svg?style=flat-square)](https://www.npmjs.com/package/keycode-js)
![GitHub contributors](https://img.shields.io/github/contributors/kabirbaidhya/keycode-js.svg?style=flat-square)
[![PRs Welcome](https://img.shields.io/badge/PRs-welcome-brightgreen.svg?style=flat-square)](#contributing)
[![License: MIT](https://img.shields.io/badge/License-MIT-yellow.svg?style=flat-square)](LICENSE)

A minimal JavaScript package with key code constants.

## Installation

```bash
# Using npm
$ npm install keycode-js --save

# Using yarn
$ yarn add keycode-js

# Using bower
$ bower install keycode-js --save
```

## Usage

Import the package using `require()` or ES / TypeScript `import` statement:

```js
// CommonJS
const KeyCode = require('keycode-js');

// ES2015+
import * as KeyCode from 'keycode-js';

// TypeScript
import * as KeyCode from 'keycode-js';

// Import individual constants
import { KEY_RETURN } from 'keycode-js';
```

Or, using `<script>` tag:

```html
<script src="https://unpkg.com/keycode-js@3.1.0/dist/keycode.min.js"></script>
```

Finally:

```js
window.addEventListener('keyup', function(e) {
  // You may do one of these checks.

  // Check the code value.
  if (e.code === KeyCode.CODE_RETURN) {
    console.log('It was the Return key.');
    return;
  }

  // OR, check the keyCode value.
  if (e.keyCode === KeyCode.KEY_RETURN) {
    console.log('It was the Return key.');
    return;
  }

  // OR, check the key value.
  if (e.key === KeyCode.VALUE_RETURN) {
    console.log('It was the Return key.');
    return;
  }

  console.log('It was any other key.');
});
```

## Usage with Deno

```ts
// Import all constants
import * as KeyCode from 'https://deno.land/x/keycode@v3.1.0/mod.ts';

// Import individual constants
import { KEY_ENTER } from 'https://deno.land/x/keycode@v3.1.0/mod.ts';

console.assert(KeyCode.KEY_ENTER === KEY_ENTER);
```

## Available Constants

### 1. Key event code constants

List of available constants that corresponds to the browser's [`KeyboardEvent.code`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code) attribute.

[[Reference](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/code/code_values)]


**Example usage**

```js
window.addEventListener('keyup', function(e) {
  console.log('Enter Key:', (e.code === KeyCode.CODE_RETURN));
});
```

**List**

| Key Name               | Constant                   | Value              |
|------------------------|----------------------------|--------------------|
| (Undefined)            | CODE_UNIDENTIFIED          | `"Unidentified"`   |
| Escape                 | CODE_ESCAPE                | `"Escape"`         |
| Dash / Minus           | CODE_MINUS<br/>CODE_DASH   | `"Minus"`          |
| Equals                 | CODE_EQUALS                | `"Equal"`          |
| Backspace              | CODE_BACK_SPACE            | `"Backspace"`      |
| Tab                    | CODE_TAB                   | `"Tab"`            |
| Return                 | CODE_ENTER<br/>CODE_RETURN | `"Enter"`          |
| Shift (Left)           | CODE_SHIFT_LEFT            | `"ShiftLeft"`      |
| Shift (Right)          | CODE_SHIFT_RIGHT           | `"ShiftRight"`     |
| Control (Left)         | CODE_CONTROL_LEFT          | `"ControlLeft"`    |
| Control (Right)        | CODE_CONTROL_RIGHT         | `"ControlRight"`   |
| Alt (Left)             | CODE_ALT_LEFT              | `"AltLeft"`        |
| Alt (Right)            | CODE_ALT_RIGHT             | `"AltRight"`       |
| Pause                  | CODE_PAUSE                 | `"Pause"`          |
| Caps Lock              | CODE_CAPS_LOCK             | `"CapsLock"`       |
| Space                  | CODE_SPACE                 | `"Space"`          |
| Page Up                | CODE_PAGE_UP               | `"PageUp"`         |
| Page Down              | CODE_PAGE_DOWN             | `"PageDown"`       |
| End                    | CODE_END                   | `"End"`            |
| Home                   | CODE_HOME                  | `"Home"`           |
| Left                   | CODE_LEFT                  | `"ArrowLeft"`      |
| Up                     | CODE_UP                    | `"ArrowUp"`        |
| Right                  | CODE_RIGHT                 | `"ArrowRight"`     |
| Down                   | CODE_DOWN                  | `"ArrowDown"`      |
| Print Screen           | CODE_PRINTSCREEN           | `"PrintScreen"`    |
| Insert                 | CODE_INSERT                | `"Insert"`         |
| Delete                 | CODE_DELETE                | `"Delete"`         |
| 0                      | CODE_0                     | `"Digit0"`         |
| 1                      | CODE_1                     | `"Digit1"`         |
| 2                      | CODE_2                     | `"Digit2"`         |
| 3                      | CODE_3                     | `"Digit3"`         |
| 4                      | CODE_4                     | `"Digit4"`         |
| 5                      | CODE_5                     | `"Digit5"`         |
| 6                      | CODE_6                     | `"Digit6"`         |
| 7                      | CODE_7                     | `"Digit7"`         |
| 8                      | CODE_8                     | `"Digit8"`         |
| 9                      | CODE_9                     | `"Digit9"`         |
| A                      | CODE_A                     | `"KeyA"`           |
| B                      | CODE_B                     | `"KeyB"`           |
| C                      | CODE_C                     | `"KeyC"`           |
| D                      | CODE_D                     | `"KeyD"`           |
| E                      | CODE_E                     | `"KeyE"`           |
| F                      | CODE_F                     | `"KeyF"`           |
| G                      | CODE_G                     | `"KeyG"`           |
| H                      | CODE_H                     | `"KeyH"`           |
| I                      | CODE_I                     | `"KeyI"`           |
| J                      | CODE_J                     | `"KeyJ"`           |
| K                      | CODE_K                     | `"KeyK"`           |
| L                      | CODE_L                     | `"KeyL"`           |
| M                      | CODE_M                     | `"KeyM"`           |
| N                      | CODE_N                     | `"KeyN"`           |
| O                      | CODE_O                     | `"KeyO"`           |
| P                      | CODE_P                     | `"KeyP"`           |
| Q                      | CODE_Q                     | `"KeyQ"`           |
| R                      | CODE_R                     | `"KeyR"`           |
| S                      | CODE_S                     | `"KeyS"`           |
| T                      | CODE_T                     | `"KeyT"`           |
| U                      | CODE_U                     | `"KeyU"`           |
| V                      | CODE_V                     | `"KeyV"`           |
| W                      | CODE_W                     | `"KeyW"`           |
| X                      | CODE_X                     | `"KeyX"`           |
| Y                      | CODE_Y                     | `"KeyY"`           |
| Z                      | CODE_Z                     | `"KeyZ"`           |
| Cmd ⌘ / Window (Left)  | CODE_META_LEFT             | `"MetaLeft"`       |
| Cmd ⌘ / Window (Left)  | CODE_OS_LEFT               | `"OSLeft"`         |
| Cmd ⌘ / Window (Right) | CODE_META_RIGHT            | `"MetaRight"`      |
| Cmd ⌘ / Window (Right) | CODE_OS_RIGHT              | `"OSRight"`        |
| Context Menu           | CODE_CONTEXT_MENU          | `"ContextMenu"`    |
| Numpad 0               | CODE_NUMPAD0               | `"Numpad0"`        |
| Numpad 1               | CODE_NUMPAD1               | `"Numpad1"`        |
| Numpad 2               | CODE_NUMPAD2               | `"Numpad2"`        |
| Numpad 3               | CODE_NUMPAD3               | `"Numpad3"`        |
| Numpad 4               | CODE_NUMPAD4               | `"Numpad4"`        |
| Numpad 5               | CODE_NUMPAD5               | `"Numpad5"`        |
| Numpad 6               | CODE_NUMPAD6               | `"Numpad6"`        |
| Numpad 7               | CODE_NUMPAD7               | `"Numpad7"`        |
| Numpad 8               | CODE_NUMPAD8               | `"Numpad8"`        |
| Numpad 9               | CODE_NUMPAD9               | `"Numpad9"`        |
| Multiply               | CODE_NUMPAD_MULTIPLY       | `"NumpadMultiply"` |
| Add                    | CODE_NUMPAD_ADD            | `"NumpadAdd"`      |
| Subtract               | CODE_NUMPAD_SUBTRACT       | `"NumpadSubtract"` |
| Decimal                | CODE_NUMPAD_DECIMAL        | `"NumpadDecimal"`  |
| Divide                 | CODE_NUMPAD_DIVIDE         | `"NumpadDivide"`   |
| Numpad Enter           | CODE_NUMPAD_ENTER          | `"NumpadEnter"`    |
| F1                     | CODE_F1                    | `"F1"`             |
| F2                     | CODE_F2                    | `"F2"`             |
| F3                     | CODE_F3                    | `"F3"`             |
| F4                     | CODE_F4                    | `"F4"`             |
| F5                     | CODE_F5                    | `"F5"`             |
| F6                     | CODE_F6                    | `"F6"`             |
| F7                     | CODE_F7                    | `"F7"`             |
| F8                     | CODE_F8                    | `"F8"`             |
| F9                     | CODE_F9                    | `"F9"`             |
| F10                    | CODE_F10                   | `"F10"`            |
| F11                    | CODE_F11                   | `"F11"`            |
| F12                    | CODE_F12                   | `"F12"`            |
| F13                    | CODE_F13                   | `"F13"`            |
| F14                    | CODE_F14                   | `"F14"`            |
| F15                    | CODE_F15                   | `"F15"`            |
| F16                    | CODE_F16                   | `"F16"`            |
| F17                    | CODE_F17                   | `"F17"`            |
| F18                    | CODE_F18                   | `"F18"`            |
| F19                    | CODE_F19                   | `"F19"`            |
| F20                    | CODE_F20                   | `"F20"`            |
| F21                    | CODE_F21                   | `"F21"`            |
| F22                    | CODE_F22                   | `"F22"`            |
| F23                    | CODE_F23                   | `"F23"`            |
| F24                    | CODE_F24                   | `"F24"`            |
| Num Lock               | CODE_NUM_LOCK              | `"NumLock"`        |
| Scroll Lock            | CODE_SCROLL_LOCK           | `"ScrollLock"`     |
| Semicolon              | CODE_SEMICOLON             | `"Semicolon"`      |
| Comma                  | CODE_COMMA                 | `"Comma"`          |
| Period                 | CODE_PERIOD                | `"Period"`         |
| Slash                  | CODE_SLASH                 | `"Slash"`          |
| Back Quote             | CODE_BACK_QUOTE            | `"Backquote"`      |
| Open Bracket           | CODE_OPEN_BRACKET          | `"BracketLeft"`    |
| Back Slash             | CODE_BACK_SLASH            | `"Backslash"`      |
| Close Bracket          | CODE_CLOSE_BRACKET         | `"BracketRight"`   |
| Quote                  | CODE_QUOTE                 | `"Quote"`          |

### 2. Key code constants (numeric)

List of available numeric constants that corresponds to the browser's [`KeyboardEvent.keyCode`](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode) attribute.

[[Reference](https://developer.mozilla.org/en-US/docs/Web/API/KeyboardEvent/keyCode#Constants_for_keyCode_value)]

**Example usage**

```js
window.addEventListener('keyup', function(e) {

CHANGELOG.md
# Change Log

## [v3.1.0](https://github.com/kabirbaidhya/keycode-js/tree/v3.1.0) (2020-06-14)
[Full Changelog](https://github.com/kabirbaidhya/keycode-js/compare/v3.0.0...v3.1.0)

**Changes**

- Add key event code constants [\#38](https://github.com/kabirbaidhya/keycode-js/pull/38) [[feature](https://github.com/kabirbaidhya/keycode-js/labels/feature)] ([kabirbaidhya](https://github.com/kabirbaidhya))
- Add key value constants [\#37](https://github.com/kabirbaidhya/keycode-js/pull/37) [[feature](https://github.com/kabirbaidhya/keycode-js/labels/feature)] ([kabirbaidhya](https://github.com/kabirbaidhya))

## [v3.0.0](https://github.com/kabirbaidhya/keycode-js/tree/v3.0.0) (2020-06-05)
[Full Changelog](https://github.com/kabirbaidhya/keycode-js/compare/v2.0.3...v3.0.0)

**Closed Issues**

- Missing keycodes [\#35](https://github.com/kabirbaidhya/keycode-js/issues/35) [[feature](https://github.com/kabirbaidhya/keycode-js/labels/feature)]

**Changes**

- Update missing keys and prefix key codes specific to firefox [\#36](https://github.com/kabirbaidhya/keycode-js/pull/36) [[breaking change](https://github.com/kabirbaidhya/keycode-js/labels/breaking%20change)] ([kabirbaidhya](https://github.com/kabirbaidhya))
- Update outdated dev dependencies [\#33](https://github.com/kabirbaidhya/keycode-js/pull/33) [[dependencies](https://github.com/kabirbaidhya/keycode-js/labels/dependencies)] ([kabirbaidhya](https://github.com/kabirbaidhya))

## [v2.0.3](https://github.com/kabirbaidhya/keycode-js/tree/v2.0.3) (2019-09-02)
[Full Changelog](https://github.com/kabirbaidhya/keycode-js/compare/v2.0.2...v2.0.3)

**Changes**

- Generate ES5 compatible build [\#32](https://github.com/kabirbaidhya/keycode-js/pull/32) ([inspirado](https://github.com/inspirado))

## [v2.0.2](https://github.com/kabirbaidhya/keycode-js/tree/v2.0.2) (2019-08-27)
[Full Changelog](https://github.com/kabirbaidhya/keycode-js/compare/v2.0.1...v2.0.2)

**Changes**

- Make the main build \(common js\) fully-compatible with ES5 [\#31](https://github.com/kabirbaidhya/keycode-js/pull/31) [[bugfix](https://github.com/kabirbaidhya/keycode-js/labels/bugfix)] ([kabirbaidhya](https://github.com/kabirbaidhya))

## [v2.0.1](https://github.com/kabirbaidhya/keycode-js/tree/v2.0.1) (2019-08-20)
[Full Changelog](https://github.com/kabirbaidhya/keycode-js/compare/v2.0.0...v2.0.1)

**Changes**

- Use named export for ES2015 [\#29](https://github.com/kabirbaidhya/keycode-js/pull/29) [[documentation](https://github.com/kabirbaidhya/keycode-js/labels/documentation)] ([mesaugat](https://github.com/mesaugat))

## [v2.0.0](https://github.com/kabirbaidhya/keycode-js/tree/v2.0.0) (2019-08-20)
[Full Changelog](https://github.com/kabirbaidhya/keycode-js/compare/v1.0.4...v2.0.0)

**Closed Issues**

- Add an example for usage with deno [\#26](https://github.com/kabirbaidhya/keycode-js/issues/26) [[example](https://github.com/kabirbaidhya/keycode-js/labels/example)] [[good first issue](https://github.com/kabirbaidhya/keycode-js/labels/good%20first%20issue)] [[help wanted](https://github.com/kabirbaidhya/keycode-js/labels/help%20wanted)]
- Make sure everything works when used in browser via a script tag [\#25](https://github.com/kabirbaidhya/keycode-js/issues/25) [[test](https://github.com/kabirbaidhya/keycode-js/labels/test)]
- Add support for deno i.e make it available on deno registry [\#21](https://github.com/kabirbaidhya/keycode-js/issues/21) [[feature](https://github.com/kabirbaidhya/keycode-js/labels/feature)]

**Changes**

- Test and build enhancements for node, deno and browser builds [\#28](https://github.com/kabirbaidhya/keycode-js/pull/28) [[test](https://github.com/kabirbaidhya/keycode-js/labels/test)] ([kabirbaidhya](https://github.com/kabirbaidhya))
- Include docs for usage with deno [\#27](https://github.com/kabirbaidhya/keycode-js/pull/27) [[documentation](https://github.com/kabirbaidhya/keycode-js/labels/documentation)] ([mesaugat](https://github.com/mesaugat))
- Make keycode-js compatible with deno [\#24](https://github.com/kabirbaidhya/keycode-js/pull/24) [[breaking change](https://github.com/kabirbaidhya/keycode-js/labels/breaking%20change)] [[feature](https://github.com/kabirbaidhya/keycode-js/labels/feature)] [[refactor](https://github.com/kabirbaidhya/keycode-js/labels/refactor)] ([mesaugat](https://github.com/mesaugat))
- Change unpkg URL to next version when creating changelog [\#23](https://github.com/kabirbaidhya/keycode-js/pull/23) ([mesaugat](https://github.com/mesaugat))
- Remove deprecated options from rollup config [\#22](https://github.com/kabirbaidhya/keycode-js/pull/22) ([mesaugat](https://github.com/mesaugat))
- Run tests for both original library and the generated build  [\#20](https://github.com/kabirbaidhya/keycode-js/pull/20) [[test](https://github.com/kabirbaidhya/keycode-js/labels/test)] ([kabirbaidhya](https://github.com/kabirbaidhya))
- Use eslint-config-leapfrog for eslint configuration [\#19](https://github.com/kabirbaidhya/keycode-js/pull/19) [[dependencies](https://github.com/kabirbaidhya/keycode-js/labels/dependencies)] ([kabirbaidhya](https://github.com/kabirbaidhya))
- Remove npm-run-all and other unused dependencies [\#18](https://github.com/kabirbaidhya/keycode-js/pull/18) [[dependencies](https://github.com/kabirbaidhya/keycode-js/labels/dependencies)] ([kabirbaidhya](https://github.com/kabirbaidhya))
- Bump lodash from 4.17.10 to 4.17.14 [\#17](https://github.com/kabirbaidhya/keycode-js/pull/17) [[dependencies](https://github.com/kabirbaidhya/keycode-js/labels/dependencies)] ([dependabot[bot]](https://github.com/apps/dependabot))

## [v1.0.4](https://github.com/kabirbaidhya/keycode-js/tree/v1.0.4) (2019-01-31)
[Full Changelog](https://github.com/kabirbaidhya/keycode-js/compare/v1.0.3...v1.0.4)

**Implemented enhancements:**


mod.d.ts
export declare const KEY_CANCEL = 3;
export declare const KEY_HELP = 6;
export declare const KEY_BACK_SPACE = 8;
export declare const KEY_TAB = 9;
export declare const KEY_CLEAR = 12;
export declare const KEY_RETURN = 13;
export declare const KEY_SHIFT = 16;
export declare const KEY_CONTROL = 17;
export declare const KEY_ALT = 18;
export declare const KEY_PAUSE = 19;
export declare const KEY_CAPS_LOCK = 20;
export declare const KEY_ESCAPE = 27;
export declare const KEY_SPACE = 32;
export declare const KEY_PAGE_UP = 33;
export declare const KEY_PAGE_DOWN = 34;
export declare const KEY_END = 35;
export declare const KEY_HOME = 36;
export declare const KEY_LEFT = 37;
export declare const KEY_UP = 38;
export declare const KEY_RIGHT = 39;
export declare const KEY_DOWN = 40;
export declare const KEY_PRINTSCREEN = 44;
export declare const KEY_INSERT = 45;
export declare const KEY_DELETE = 46;
export declare const KEY_0 = 48;
export declare const KEY_1 = 49;
export declare const KEY_2 = 50;
export declare const KEY_3 = 51;
export declare const KEY_4 = 52;
export declare const KEY_5 = 53;
export declare const KEY_6 = 54;
export declare const KEY_7 = 55;
export declare const KEY_8 = 56;
export declare const KEY_9 = 57;
export declare const KEY_A = 65;
export declare const KEY_B = 66;
export declare const KEY_C = 67;
export declare const KEY_D = 68;
export declare const KEY_E = 69;
export declare const KEY_F = 70;
export declare const KEY_G = 71;
export declare const KEY_H = 72;
export declare const KEY_I = 73;
export declare const KEY_J = 74;
export declare const KEY_K = 75;
export declare const KEY_L = 76;
export declare const KEY_M = 77;
export declare const KEY_N = 78;
export declare const KEY_O = 79;
export declare const KEY_P = 80;
export declare const KEY_Q = 81;
export declare const KEY_R = 82;
export declare const KEY_S = 83;
export declare const KEY_T = 84;
export declare const KEY_U = 85;
export declare const KEY_V = 86;
export declare const KEY_W = 87;
export declare const KEY_X = 88;
export declare const KEY_Y = 89;
export declare const KEY_Z = 90;
export declare const KEY_LEFT_CMD = 91;
export declare const KEY_RIGHT_CMD = 92;
export declare const KEY_CONTEXT_MENU = 93;
export declare const KEY_NUMPAD0 = 96;
export declare const KEY_NUMPAD1 = 97;
export declare const KEY_NUMPAD2 = 98;
export declare const KEY_NUMPAD3 = 99;
export declare const KEY_NUMPAD4 = 100;
export declare const KEY_NUMPAD5 = 101;
export declare const KEY_NUMPAD6 = 102;
export declare const KEY_NUMPAD7 = 103;
export declare const KEY_NUMPAD8 = 104;
export declare const KEY_NUMPAD9 = 105;
export declare const KEY_MULTIPLY = 106;
export declare const KEY_ADD = 107;
export declare const KEY_SUBTRACT = 109;
export declare const KEY_DECIMAL = 110;
export declare const KEY_DIVIDE = 111;
export declare const KEY_F1 = 112;
export declare const KEY_F2 = 113;
export declare const KEY_F3 = 114;
export declare const KEY_F4 = 115;
export declare const KEY_F5 = 116;
export declare const KEY_F6 = 117;
export declare const KEY_F7 = 118;
export declare const KEY_F8 = 119;
export declare const KEY_F9 = 120;
export declare const KEY_F10 = 121;
export declare const KEY_F11 = 122;
export declare const KEY_F12 = 123;
export declare const KEY_F13 = 124;
export declare const KEY_F14 = 125;
export declare const KEY_F15 = 126;
export declare const KEY_F16 = 127;
export declare const KEY_F17 = 128;
export declare const KEY_F18 = 129;
export declare const KEY_F19 = 130;
export declare const KEY_F20 = 131;
export declare const KEY_F21 = 132;
export declare const KEY_F22 = 133;
export declare const KEY_F23 = 134;
export declare const KEY_F24 = 135;
export declare const KEY_NUM_LOCK = 144;
export declare const KEY_SCROLL_LOCK = 145;
export declare const KEY_SEMICOLON = 186;
export declare const KEY_EQUALS = 187;
export declare const KEY_COMMA = 188;
export declare const KEY_DASH = 189;
export declare const KEY_PERIOD = 190;
export declare const KEY_SLASH = 191;
export declare const KEY_BACK_QUOTE = 192;
export declare const KEY_OPEN_BRACKET = 219;
export declare const KEY_BACK_SLASH = 220;
export declare const KEY_CLOSE_BRACKET = 221;
export declare const KEY_QUOTE = 222;
export declare const KEY_FIREFOX_ENTER = 14;
export declare const KEY_FIREFOX_SEMICOLON = 59;
export declare const KEY_FIREFOX_EQUALS = 61;
export declare const KEY_FIREFOX_SEPARATOR = 108;
export declare const KEY_FIREFOX_META = 224;
export declare const VALUE_CANCEL = "Cancel";
export declare const VALUE_HELP = "Help";
export declare const VALUE_BACK_SPACE = "Backspace";
export declare const VALUE_TAB = "Tab";
export declare const VALUE_CLEAR = "Clear";
export declare const VALUE_ENTER = "Enter";
export declare const VALUE_RETURN = "Enter";
export declare const VALUE_SHIFT = "Shift";
export declare const VALUE_CONTROL = "Control";
export declare const VALUE_ALT = "Alt";
export declare const VALUE_PAUSE = "Pause";
export declare const VALUE_CAPS_LOCK = "CapsLock";
export declare const VALUE_ESCAPE = "Escape";
export declare const VALUE_SPACE = " ";
export declare const VALUE_PAGE_UP = "PageUp";
export declare const VALUE_PAGE_DOWN = "PageDown";
export declare const VALUE_END = "End";
export declare const VALUE_HOME = "Home";
export declare const VALUE_LEFT = "ArrowLeft";
export declare const VALUE_UP = "ArrowUp";
export declare const VALUE_RIGHT = "ArrowRight";
export declare const VALUE_DOWN = "ArrowDown";
export declare const VALUE_PRINTSCREEN = "PrintScreen";
export declare const VALUE_INSERT = "Insert";
export declare const VALUE_DELETE = "Delete";
export declare const VALUE_0 = "0";
export declare const VALUE_1 = "1";
export declare const VALUE_2 = "2";
export declare const VALUE_3 = "3";
export declare const VALUE_4 = "4";
export declare const VALUE_5 = "5";
export declare const VALUE_6 = "6";
export declare const VALUE_7 = "7";
export declare const VALUE_8 = "8";
export declare const VALUE_9 = "9";
export declare const VALUE_A = "a";
export declare const VALUE_B = "b";
export declare const VALUE_C = "c";
export declare const VALUE_D = "d";
export declare const VALUE_E = "e";
export declare const VALUE_F = "f";
export declare const VALUE_G = "g";
export declare const VALUE_H = "h";
export declare const VALUE_I = "i";
export declare const VALUE_J = "j";
export declare const VALUE_K = "k";
export declare const VALUE_L = "l";
export declare const VALUE_M = "m";
export declare const VALUE_N = "n";
export declare const VALUE_O = "o";
export declare const VALUE_P = "p";
export declare const VALUE_Q = "q";
export declare const VALUE_R = "r";
export declare const VALUE_S = "s";
export declare const VALUE_T = "t";
export declare const VALUE_U = "u";
export declare const VALUE_V = "v";
export declare const VALUE_W = "w";
export declare const VALUE_X = "x";
export declare const VALUE_Y = "y";
export declare const VALUE_Z = "z";
export declare const VALUE_META = "Meta";
export declare const VALUE_LEFT_CMD = "Meta";
export declare const VALUE_RIGHT_CMD = "Meta";
export declare const VALUE_CONTEXT_MENU = "ContextMenu";
export declare const VALUE_NUMPAD0 = "0";
export declare const VALUE_NUMPAD1 = "1";
export declare const VALUE_NUMPAD2 = "2";
export declare const VALUE_NUMPAD3 = "3";
export declare const VALUE_NUMPAD4 = "4";
export declare const VALUE_NUMPAD5 = "5";
export declare const VALUE_NUMPAD6 = "6";
export declare const VALUE_NUMPAD7 = "7";
export declare const VALUE_NUMPAD8 = "8";
export declare const VALUE_NUMPAD9 = "9";
export declare const VALUE_MULTIPLY = "*";
export declare const VALUE_ADD = "+";
export declare const VALUE_SUBTRACT = "-";
export declare const VALUE_DECIMAL = ".";
export declare const VALUE_DIVIDE = "/";
export declare const VALUE_F1 = "F1";
export declare const VALUE_F2 = "F2";
export declare const VALUE_F3 = "F3";
export declare const VALUE_F4 = "F4";
export declare const VALUE_F5 = "F5";
export declare const VALUE_F6 = "F6";
export declare const VALUE_F7 = "F7";
export declare const VALUE_F8 = "F8";
export declare const VALUE_F9 = "F9";
export declare const VALUE_F10 = "F10";
export declare const VALUE_F11 = "F11";
export declare const VALUE_F12 = "F12";
export declare const VALUE_F13 = "F13";
export declare const VALUE_F14 = "F14";
export declare const VALUE_F15 = "F15";
export declare const VALUE_F16 = "F16";
export declare const VALUE_F17 = "F17";
export declare const VALUE_F18 = "F18";
export declare const VALUE_F19 = "F19";
export declare const VALUE_F20 = "F20";
export declare const VALUE_F21 = "F21";
export declare const VALUE_F22 = "F22";
export declare const VALUE_F23 = "F23";
export declare const VALUE_F24 = "F24";
export declare const VALUE_NUM_LOCK = "NumLock";
export declare const VALUE_SCROLL_LOCK = "ScrollLock";
export declare const VALUE_SEMICOLON = ";";
export declare const VALUE_EQUALS = "=";
export declare const VALUE_COMMA = ",";
export declare const VALUE_DASH = "-";
export declare const VALUE_PERIOD = ".";
export declare const VALUE_SLASH = "/";
export declare const VALUE_BACK_QUOTE = "`";
export declare const VALUE_OPEN_BRACKET = "[";
export declare const VALUE_BACK_SLASH = "\\";
export declare const VALUE_CLOSE_BRACKET = "]";
export declare const VALUE_QUOTE = "'";
export declare const CODE_UNIDENTIFIED = "Unidentified";
export declare const CODE_ESCAPE = "Escape";
export declare const CODE_MINUS = "Minus";
export declare const CODE_DASH = "Minus";
export declare const CODE_EQUALS = "Equal";
export declare const CODE_BACK_SPACE = "Backspace";
export declare const CODE_TAB = "Tab";
export declare const CODE_ENTER = "Enter";
export declare const CODE_RETURN = "Enter";
export declare const CODE_SHIFT_LEFT = "ShiftLeft";
export declare const CODE_SHIFT_RIGHT = "ShiftRight";
export declare const CODE_CONTROL_LEFT = "ControlLeft";
export declare const CODE_CONTROL_RIGHT = "ControlRight";
export declare const CODE_ALT_LEFT = "AltLeft";
export declare const CODE_ALT_RIGHT = "AltRight";
export declare const CODE_PAUSE = "Pause";
export declare const CODE_CAPS_LOCK = "CapsLock";
export declare const CODE_SPACE = "Space";
export declare const CODE_PAGE_UP = "PageUp";
export declare const CODE_PAGE_DOWN = "PageDown";
export declare const CODE_END = "End";
export declare const CODE_HOME = "Home";
export declare const CODE_LEFT = "ArrowLeft";
export declare const CODE_UP = "ArrowUp";
export declare const CODE_RIGHT = "ArrowRight";
export declare const CODE_DOWN = "ArrowDown";
export declare const CODE_PRINTSCREEN = "PrintScreen";
export declare const CODE_INSERT = "Insert";
export declare const CODE_DELETE = "Delete";
export declare const CODE_0 = "Digit0";
export declare const CODE_1 = "Digit1";
export declare const CODE_2 = "Digit2";
export declare const CODE_3 = "Digit3";
export declare const CODE_4 = "Digit4";
export declare const CODE_5 = "Digit5";
export declare const CODE_6 = "Digit6";
export declare const CODE_7 = "Digit7";
export declare const CODE_8 = "Digit8";
export declare const CODE_9 = "Digit9";
export declare const CODE_A = "KeyA";
export declare const CODE_B = "KeyB";
export declare const CODE_C = "KeyC";
export declare const CODE_D = "KeyD";
export declare const CODE_E = "KeyE";
export declare const CODE_F = "KeyF";
export declare const CODE_G = "KeyG";
export declare const CODE_H = "KeyH";
export declare const CODE_I = "KeyI";
export declare const CODE_J = "KeyJ";
export declare const CODE_K = "KeyK";
export declare const CODE_L = "KeyL";
export declare const CODE_M = "KeyM";
export declare const CODE_N = "KeyN";
export declare const CODE_O = "KeyO";
export declare const CODE_P = "KeyP";
export declare const CODE_Q = "KeyQ";
export declare const CODE_R = "KeyR";
export declare const CODE_S = "KeyS";
export declare const CODE_T = "KeyT";
export declare const CODE_U = "KeyU";
export declare const CODE_V = "KeyV";
export declare const CODE_W = "KeyW";
export declare const CODE_X = "KeyX";
export declare const CODE_Y = "KeyY";
export declare const CODE_Z = "KeyZ";
export declare const CODE_META_LEFT = "MetaLeft";
export declare const CODE_OS_LEFT = "OSLeft";
export declare const CODE_META_RIGHT = "MetaRight";
export declare const CODE_OS_RIGHT = "OSRight";
export declare const CODE_CONTEXT_MENU = "ContextMenu";
export declare const CODE_NUMPAD0 = "Numpad0";
export declare const CODE_NUMPAD1 = "Numpad1";
export declare const CODE_NUMPAD2 = "Numpad2";
export declare const CODE_NUMPAD3 = "Numpad3";
export declare const CODE_NUMPAD4 = "Numpad4";
export declare const CODE_NUMPAD5 = "Numpad5";
export declare const CODE_NUMPAD6 = "Numpad6";
export declare const CODE_NUMPAD7 = "Numpad7";
export declare const CODE_NUMPAD8 = "Numpad8";
export declare const CODE_NUMPAD9 = "Numpad9";
export declare const CODE_NUMPAD_MULTIPLY = "NumpadMultiply";
export declare const CODE_NUMPAD_ADD = "NumpadAdd";
export declare const CODE_NUMPAD_SUBTRACT = "NumpadSubtract";
export declare const CODE_NUMPAD_DECIMAL = "NumpadDecimal";
export declare const CODE_NUMPAD_DIVIDE = "NumpadDivide";
export declare const CODE_NUMPAD_ENTER = "NumpadEnter";
export declare const CODE_F1 = "F1";
export declare const CODE_F2 = "F2";
export declare const CODE_F3 = "F3";
export declare const CODE_F4 = "F4";
export declare const CODE_F5 = "F5";
export declare const CODE_F6 = "F6";
export declare const CODE_F7 = "F7";
export declare const CODE_F8 = "F8";
export declare const CODE_F9 = "F9";
export declare const CODE_F10 = "F10";
export declare const CODE_F11 = "F11";
export declare const CODE_F12 = "F12";
export declare const CODE_F13 = "F13";
export declare const CODE_F14 = "F14";
export declare const CODE_F15 = "F15";
export declare const CODE_F16 = "F16";
export declare const CODE_F17 = "F17";
export declare const CODE_F18 = "F18";
export declare const CODE_F19 = "F19";
export declare const CODE_F20 = "F20";
export declare const CODE_F21 = "F21";
export declare const CODE_F22 = "F22";
export declare const CODE_F23 = "F23";
export declare const CODE_F24 = "F24";
export declare const CODE_NUM_LOCK = "NumLock";
export declare const CODE_SCROLL_LOCK = "ScrollLock";
export declare const CODE_SEMICOLON = "Semicolon";
export declare const CODE_COMMA = "Comma";
export declare const CODE_PERIOD = "Period";
export declare const CODE_SLASH = "Slash";
export declare const CODE_BACK_QUOTE = "Backquote";
export declare const CODE_OPEN_BRACKET = "BracketLeft";
export declare const CODE_BACK_SLASH = "Backslash";
export declare const CODE_CLOSE_BRACKET = "BracketRight";
export declare const CODE_QUOTE = "Quote";
